rusty-machine/src/learning/nnet.rs

```diff
@@ -211,13 +211,13 @@ impl<'a, T: Criterion> NeuralNet<'a, T> {
     */
 
     /// Compute the gradient using the back propagation algorithm.
-    fn compute_grad(&self, weights: &[f64], data: &Matrix<f64>, outputs: &Matrix<f64>) -> (f64, Vec<f64>) {
-        assert_eq!(data.cols(), self.layer_sizes[0]);
+    fn compute_grad(&self, weights: &[f64], inputs: &Matrix<f64>, outputs: &Matrix<f64>) -> (f64, Vec<f64>) {
```

Comment:
`outputs` in this signature should be `targets` to match the compute_grad function below.